oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsGraphicsCard.java

```diff
@@ -49,6 +66,76 @@ final class WindowsGraphicsCard extends AbstractGraphicsCard {
      */
     public static List<GraphicsCard> getGraphicsCards() {
         List<GraphicsCard> cardList = new ArrayList<>();
+
+        String[] keys = Advapi32Util.registryGetKeys(WinReg.HKEY_LOCAL_MACHINE, DISPLAY_DEVICES_REGISTRY_PATH);
+        for (String key : keys) {
+            if (!key.startsWith("0")) {
+                continue;
+            }
+
+            try {
+                String fullKey = DISPLAY_DEVICES_REGISTRY_PATH + key;
+                if (!Advapi32Util.registryValueExists(WinReg.HKEY_LOCAL_MACHINE, fullKey, ADAPTER_STRING)) {
+                    continue;
+                }
+
+                String name = Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, DRIVER_DESC);
+                String deviceId = null; // Note: not found proper key in registry
+                String vendor = Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, VENDOR);
+                String versionInfo = Advapi32Util.registryGetStringValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, DRIVER_VERSION);
+                long vram = 0L;
+
+                if (Advapi32Util.registryValueExists(WinReg.HKEY_LOCAL_MACHINE, fullKey, QW_MEMORY_SIZE)) {
+                    vram = Advapi32Util.registryGetLongValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, QW_MEMORY_SIZE);
+                } else if (Advapi32Util.registryValueExists(WinReg.HKEY_LOCAL_MACHINE, fullKey, MEMORY_SIZE)) {
+                    Object genericValue = Advapi32Util.registryGetValue(WinReg.HKEY_LOCAL_MACHINE, fullKey, MEMORY_SIZE);
+                    if (genericValue instanceof Long || genericValue instanceof Integer) {
+                        vram = (long) genericValue;
```

Comment:
I think you need to handle `Long` and `Integer` separately as the int will be an unsigned DWORD.
```suggestion
                    if (genericValue instanceof Long) {
                        vram = genericValue.longValue();
                    else if (genericValue instanceof Integer) {
                        vram = Integer.toUnsignedLong(genericValue.intValue());
```